CodeGen/Generators/QuantityJsonFilesParser.cs

```diff
@@ -124,5 +126,254 @@ private static Localization[] GetLocalizationForPrefixUnit(IEnumerable<Localizat
+        /// </summary>
+        /// <param name="dimensions">SI base unit dimensions of quantity, e.g. L=1 for Length or L=1,T=-1 for Speed.</param>
+        /// <param name="baseUnits">SI base units for a non-prefixed unit, e.g. L=Meter for Length.Meter or L=Meter,T=Second for Speed.MeterPerSecond.</param>
+        /// <param name="prefixInfo">Information about the prefix to apply.</param>
+        /// <returns>A new instance of <paramref name="baseUnits"/> after applying the metric prefix <paramref name="prefixInfo"/>.</returns>
+        private static BaseUnits? GetPrefixedBaseUnits(BaseDimensions dimensions, BaseUnits? baseUnits, PrefixInfo prefixInfo)
+        {
+            if (baseUnits is null) return null;
+
+            // Iterate the non-zero dimension exponents in absolute-increasing order, positive first [1, -1, 2, -2...n, -n]
+            foreach (int degree in dimensions.GetNonZeroDegrees().OrderBy(int.Abs).ThenByDescending(x => x))
+            {
+                if (TryPrefixWithDegree(dimensions, baseUnits, prefixInfo.Prefix, degree, out BaseUnits? prefixedUnits))
+                {
+                    return prefixedUnits;
+                }
+            }
+
+            return null;
+        }
+
+        private static IEnumerable<int> GetNonZeroDegrees(this BaseDimensions dimensions)
+        {
+            if (dimensions.I != 0)
+            {
+                yield return dimensions.I;
+            }
+
+            if (dimensions.J != 0)
+            {
+                yield return dimensions.J;
+            }
+
+            if (dimensions.L != 0)
+            {
+                yield return dimensions.L;
+            }
+
+            if (dimensions.M != 0)
+            {
+                yield return dimensions.M;
+            }
+
+            if (dimensions.N != 0)
+            {
+                yield return dimensions.N;
+            }
+
+            if (dimensions.T != 0)
+            {
+                yield return dimensions.T;
+            }
+
+            if (dimensions.Θ != 0)
+            {
+                yield return dimensions.Θ;
+            }
+        }
+
+        /// <summary>
+        ///     TODO Explain this, in particular the degree/exponent stuff.
+        /// </summary>
+        /// <param name="dimensions"></param>
+        /// <param name="baseUnits"></param>
+        /// <param name="prefix"></param>
+        /// <param name="degree"></param>
+        /// <param name="prefixedBaseUnits"></param>
+        /// <returns></returns>
+        private static bool TryPrefixWithDegree(BaseDimensions dimensions, BaseUnits baseUnits, Prefix prefix, int degree, [NotNullWhen(true)] out BaseUnits? prefixedBaseUnits)
+        {
+            prefixedBaseUnits = baseUnits.Clone();
+
+            // look for a dimension that is part of the non-zero exponents
+            if (baseUnits.N is { } baseAmountUnit && dimensions.N == degree)
```

Comment:
I took the liberty of renaming a few vars/params for my own understanding, take a look if you agree with it
99b378f9269567018357add0ce6518d534cc1b72